frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+const { utils } = require( "ethers" );
+import { delayMsec } from "../utils/utils";
+
+import Typography from "@mui/material/Typography";
+import {
+	Button,
+	Container,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import Grid from "@mui/material/Grid";
+import ConditionalAlert from "./ConditionalAlert";
+
+function AccountEntry({ data }) {
+	let balance = "NaN";
+	// data.balance can have NaN value if no Metamask or other Wallet installed
+	if (!isNaN(data.balance)) {
+		balance = utils.formatEther(data.balance);
+	}
+	
+	return <TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{balance}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthereumAccountDetails ({ publicAddresses, setErrorNoMetamaskMessage }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		// TODO: publicAddresses is still not loaded. 
+		// Solve temporarily by waiting for 0.5 sec.
+		await delayMsec(500);
```

Comment:
Please removed this. You rarely ever want to lazily wait for something anywhere.

Reply:
You mean busily waiting, right?

Reply:
Busy waiting would be something like 
```
while (!condition) {
 // do nothing
}
// do stuff that requires condition to be true
```

I thought what I mean is called lazy waiting, but I must have been wrong. Anyways, sleeping at any point in the code (or buzy waiting) should not be done anywhere. Instead, react to something like events that are fired when a state changes, as in the case of a state chaning which results in useEffect being triggered.